workers/main.py

```
@@ -64,6 +67,73 @@ def handle_cors_preflight(origin):
+        
+        if signature_b64 != expected_sig_b64:
+            return None
+        
+        padding = '=' * (4 - len(payload_b64) % 4)
+        payload_json = base64.urlsafe_b64decode(payload_b64 + padding).decode()
+        payload = json.loads(payload_json)
+        
+        if 'exp' in payload and datetime.now().timestamp() > payload['exp']:
+            return None
+            
+        return payload
```

Ruff (0.15.4)

[warning] Consider moving this statement to an `else` block

(TRY300)